### ISSUE TITLE
Discussion: Pharo specific exercises that highlight language features

### ISSUE THREAD

**Issue by macta** [exercism/pharo-smalltalk#208]
This is placeholder to put ideas for future Pharo specific exercises that hightlight its power/features:

1. Moldability - create a scenario where users implement a #gtInspectorDetailsIn: method and see how useful it is to have custom inspector panes that also work in the debugger
1. Refactoring is not magic - have an exercise where users create a simple refactoring rule, and learn about how code is an object too, and ast's are not scary
1. The Debugger is an object - have an exercise where you make a simple change to the debugger (maybe a button to search the stack and jump to that execution point)
1. Create your own syntax - exercise to add ifMaybe: or something that would not be possibly in many languages
1. Persisting live objects - something to show how to fuel out objects and bring them back

**Comment by samWson** [comment]
Sounds like a great idea. Although the stuff you mention here is things **I** never though Smalltalk could do. I do recall there is an example of how to use Fuel in one of the free books from Pharo.org.

**Comment by RoelofWobben** [comment]
are these not exercises that could be in the hard section not in the easy one

**Comment by macta** [comment]
yes these would be more advanced (although not necessarily, there are simple things in the language as well - for example, something with points?)

**Comment by RoelofWobben** [comment]
maybe some exercises where I can practice double dispath and when I need to use that

**Comment by RoelofWobben** [comment]
or practice more with object and/or streams

**Comment by bencoman** [comment]
I added... #220 Custom exercises to introduce Reflection

**Comment by Ducasse** [comment]
In the roll die game I have a nice double dispatch exercise.
check the pdf at https://github.com/SquareBracketAssociates/PatternsOfDesign

Stef

**Comment by samWson** [comment]
@Ducasse thanks for the link to the exercise. I'll definitely have a look at it.
